Editor/Drawers/ReferenceDrawer.cs

```diff
@@ -29,10 +30,72 @@ private enum DragAndDropMode
         protected SerializedProperty RawReferenceProperty => Property.FindPropertyRelative("rawReference");
         protected SerializedProperty UnityReferenceProperty => Property.FindPropertyRelative("unityReference");
 
-        protected ReferenceDrawer(SerializedProperty property, Type genericType)
+        protected readonly FieldInfo fieldInfo;
+
+        protected ReferenceMode ModeValue
+        {
+            get => (ReferenceMode)ReferenceModeProperty.enumValueIndex;
+            set => ReferenceModeProperty.enumValueIndex = (int)value;
+        }
+
+        protected object RawReferenceValue
+        {
+            get
+            {
+#if UNITY_2021_1_OR_NEWER
+                return RawReferenceProperty.managedReferenceValue;
+#else
+                ISerializableInterface instance =
+                    (ISerializableInterface)fieldInfo.GetValue(Property.serializedObject.targetObject);
+                return instance.GetRawReference();
+#endif
+            }
+            set
+            {
+#if UNITY_2021_1_OR_NEWER
+                RawReferenceProperty.managedReferenceValue = value;
+#else
+                fieldInfo.SetValue(Property.serializedObject.targetObject, value);
+#endif
+            }
+        }
+
+        protected object PropertyValue
```

Comment:
This property mirrors the behaviour of SerializableInterface.Value, but in the PropertyDrawer.